drivers/storage/portworx/component/telemetry.go

```diff
@@ -149,712 +224,516 @@ func (t *telemetry) setTelemetryCertOwnerRef(
-        "cert_secret_namespace": "%s"
-     },
-     "cloud": {
-       "array_loc_file_path": "/etc/ccm/%s"
-     },
-      "server": {
-        "hostname": "0.0.0.0"
-      },
-      "logupload": {
-        "logfile_patterns": [
-            "/var/cores/*diags*",
-            "/var/cores/auto/*diags*",
-            "/var/cores/*px-cores*",
-            "/var/cores/*.heap",
-            "/var/cores/*.stack",
-            "/var/cores/.alerts/alerts*"
-        ],
-        "skip_patterns": [],
-        "additional_files": [
-            "/etc/pwx/config.json",
-            "/var/cores/.alerts/alerts.log",
-            "/var/cores/px_etcd_watch.log",
-            "/var/cores/px_cache_mon.log",
-            "/var/cores/px_cache_mon_watch.log",
-            "/var/cores/px_healthmon_watch.log",
-            "/var/cores/px_event_watch.log"
-        ],
-        "phonehome_sent": "/var/cache/phonehome.sent"
-      },
-      "xml_rpc": {},
-      "standalone": {
-        "version": "1.0.0",
-        "controller_sn": "SA-0",
-        "component_name": "SA-0",
-        "product_name": "portworx",
-        "appliance_id_path": "/etc/pwx/cluster_uuid"
-      },
-      "subscription": {
-        "use_appliance_id": "true"
-      },
-      "proxy": {
-        "path": "/cache/network/http_proxy"
-      }
-    }
-`, cluster.Namespace, TelemetryArcusLocationFilename)
-
-	data := map[string]string{
-		TelemetryPropertiesFilename: config,
-	}
-
-	if location, present := cluster.Annotations[pxutil.AnnotationTelemetryArcusLocation]; present && len(location) > 0 {
-		data[TelemetryArcusLocationFilename] = location
-	} else {
-		data[TelemetryArcusLocationFilename] = "external"
+	deployment, err := k8sutil.GetDeploymentFromFile(deploymentFileNameRegistrationService, pxutil.SpecsBaseDir())
+	if err != nil {
+		return err
 	}
+	telemetryImage, err := GetDesiredTelemetryImage(cluster)
+	if err != nil {
+		return err
+	}
+	proxyImage, err := getDesiredProxyImage(cluster)
+	if err != nil {
+		return err
+	}
+	var containers []v1.Container
+	for _, c := range deployment.Spec.Template.Spec.Containers {
+		if c.Name == containerNameRegistrationService {
+			c.Image = telemetryImage
+		} else if c.Name == containerNameTelemetryProxy {
+			c.Image = proxyImage
+		}
+		containers = append(containers, *c.DeepCopy())
+	}
+	deployment.Name = DeploymentNameRegistrationService
+	deployment.Namespace = cluster.Namespace
+	deployment.OwnerReferences = []metav1.OwnerReference{*ownerRef}
+	deployment.Spec.Template.Spec.Containers = containers
+	return k8sutil.CreateOrUpdateDeployment(t.k8sClient, deployment, ownerRef)
+}
 
-	return k8sutil.CreateOrUpdateConfigMap(
-		t.k8sClient,
-		&v1.ConfigMap{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:            TelemetryConfigMapName,
-				Namespace:       cluster.Namespace,
-				OwnerReferences: []metav1.OwnerReference{*ownerRef},
-			},
-			Data: data,
+func (t *telemetry) createDeploymentPhonehomeCluster(
```

Comment:
In none of the deployments, we are setting the tolerations, node affinity, image pull secret, etc. Please check other deployments for comparison.